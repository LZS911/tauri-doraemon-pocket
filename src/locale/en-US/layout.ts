export default {
  header: {
    customization: {
      title: 'Theme Customization',
      themeLayoutPanel: {
        defaultType: 'Default',
        miniType: 'Mini',
        title: 'Theme Layout',
        subTitle: 'Choose a layout type',
      },
      themeModePanel: {
        title: 'Theme Mode',
        subTitle: 'Choose a theme mode',
        light: 'Light',
        dark: 'Dark',
      },
      colorSchemePanel: {
        title: 'Color Scheme',
        subTitle: 'Choose a color scheme',
        default: 'Default',
        theme1: 'Theme1',
        theme2: 'Theme2',
      },
      fontFamilyPanel: {
        title: 'Font Family',
        subTitle: 'Choose a font family',
        default: 'default',
        inter: 'inter',
        poppins: 'poppins',
        roboto: 'roboto',
        serif: 'serif',
        mono: 'mono',
        publicSans: 'public sans',
      },
    },
    notification: {
      title: 'Notification',
      titleTips: 'Mark as all read',
      viewAll: 'View All',
    },
    userNavigation: {
      profile: {
        title: 'profile',
        editProfile: 'Edit Profile',
        viewProfile: 'View Profile',
        bindEmail: 'Bind Email',
        modifyPassword: 'Modify Password',
        singOut: 'SignOut',
      },
      settings: {
        title: 'settings',
        accountSetting: 'Account Setting',
        permissionSetting: 'Permission Setting',
      },
    },
  },
};
