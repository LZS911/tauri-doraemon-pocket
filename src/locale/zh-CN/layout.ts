export default {
  header: {
    customization: {
      title: '主题定制',
      themeLayoutPanel: {
        defaultType: '默认',
        miniType: '简略',
        title: '布局主题',
        subTitle: '选择布局类型',
      },
      themeModePanel: {
        title: '主题模式',
        subTitle: '选择主题类型',
        light: '明亮',
        dark: '暗色',
      },
      colorSchemePanel: {
        title: '配色方案',
        subTitle: '选择配色方案',
        default: '默认',
        theme1: '主题1',
        theme2: '主题2',
      },
      fontFamilyPanel: {
        title: '字体系列',
        subTitle: '选择字体',
        default: 'default',
        inter: 'inter',
        poppins: 'poppins',
        roboto: 'roboto',
        serif: 'serif',
        mono: 'mono',
        publicSans: 'public sans',
      },
    },
    notification: {
      title: '通知',
      titleTips: '标记为全部已读',
      viewAll: '查看全部',
    },
    userNavigation: {
      profile: {
        title: '简介',
        editProfile: '编辑简介',
        viewProfile: '查看简介',
        bindEmail: '邮箱绑定',
        modifyPassword: '修改密码',
        singOut: '注销',
      },
      settings: {
        title: '设置',
        accountSetting: '账号设置',
        permissionSetting: '权限设置',
      },
    },
  },
};
