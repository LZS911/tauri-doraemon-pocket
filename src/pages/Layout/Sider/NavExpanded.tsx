import { QuestionOutlined } from '@ant-design/icons';
import { Menu, MenuProps, Space, Tag } from 'antd';
import { Typography } from 'antd';
import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation, useNavigate } from 'react-router-dom';
import useTheme from '../../../customHooks/useTheme';
import { I18nKey, t as i18nT } from '../../../locale';
import { RouterConfigItem } from '../../../router';
import { routerConfig } from '../../../router/router.config';
const menuItems = (): MenuProps['items'] => {
  return routerConfig.map((v) => {
    if (v.menuType === 'Hidden') {
      return null;
    }
    if (v.children && v.label) {
      return {
        key: v.path as string,
        icon: v.icon ?? <QuestionOutlined />,
        label: <span className="text-sm">{i18nT(v.label)}</span>,
        children: v.children.map((c: RouterConfigItem) => {
          return {
            key: c.path,
            label: <span className="text-sm">{i18nT(v.label as I18nKey)}</span>,
          };
        }),
      };
    }
    return {
      key: v.path as string,
      icon: v.icon ?? <QuestionOutlined />,
      label: <span className="text-sm">{i18nT(v.label as I18nKey)}</span>,
    };
  });
};
const NavExpanded: React.FC = () => {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const location = useLocation();
  const { currentThemeMode } = useTheme();
  const handleMenuClick: MenuProps['onClick'] = (e) => {
    navigate(e.key);
  };
  const selectedKeys = useMemo(() => {
    return [location.pathname];
  }, [location]);
  return (
    <div className="flex flex-col">
      <Space direction="vertical" align="center" className="py-2">
        <img title="logo" src="/static/images/logo.svg" className="mr-3" />
        <Tag>
          <Typography.Text strong>{t('common.title')}</Typography.Text>
        </Tag>
      </Space>

      <Menu
        theme={currentThemeMode}
        onClick={handleMenuClick}
        selectedKeys={selectedKeys}
        mode="inline"
        items={menuItems()}
        className="text-base"
      />
    </div>
  );
};
export default NavExpanded;
