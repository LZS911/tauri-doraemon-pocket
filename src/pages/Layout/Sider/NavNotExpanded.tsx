import { QuestionOutlined } from '@ant-design/icons';
import { Popover } from 'antd';
import classNames from 'classnames';
import { useLocation, useNavigate } from 'react-router-dom';
import ThemeBase from '../../../components/ThemeBase';
import { genNavIconList } from './index.data';

const NavNotExpanded: React.FC = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const handleLogoClick = () => {
    navigate('dashboard');
  };

  const genNavIcon = () => {
    return genNavIconList(navigate).map((nav) => {
      if (!nav) {
        return null;
      }
      return (
        <div key={nav.name} className={`p-3`}>
          {nav.children ? (
            <Popover
              placement="rightTop"
              overlayClassName={`dark:bg-darkMode`}
              trigger="click"
              content={
                <ThemeBase.Paper className="py-2 pl-2 pr-8">
                  {nav.children.map((v) => (
                    <ThemeBase.Icon
                      data-testid={v.name}
                      isHoverCls={false}
                      className={classNames(
                        {
                          ['!text-primary']: location.pathname === v.path,
                        },
                        'mb-1',
                        'py-1',
                        'dark:bg-transparent'
                      )}
                      icon={<span>{v.name}</span>}
                      key={v.name}
                      onClick={v.onClick}
                    />
                  ))}
                </ThemeBase.Paper>
              }
            >
              <ThemeBase.Icon
                data-testid={nav.name}
                className={classNames(
                  {
                    ['!bg-secondary text-primary dark:text-primary']:
                      nav.children
                        .map((v) => v.path)
                        .includes(location.pathname),
                  },
                  'text-xl',
                  'dark:bg-darkMode'
                )}
                icon={nav.icon ?? <QuestionOutlined />}
                onClick={nav.onClick}
              />
            </Popover>
          ) : (
            <ThemeBase.Icon
              data-testid={nav.name}
              className={classNames(
                {
                  ['!bg-secondary text-primary dark:text-primary']:
                    location.pathname === nav.path,
                },
                'text-xl',
                'dark:bg-darkMode'
              )}
              icon={nav.icon ?? <QuestionOutlined />}
              onClick={nav.onClick}
            />
          )}
        </div>
      );
    });
  };

  return (
    <div className="flex flex-col">
      <img
        title="logo-img"
        data-testid="logo-img"
        src="/static/images/logo.svg"
        className="mx-auto my-2 h-10 w-10 cursor-pointer"
        onClick={handleLogoClick}
      />
      {genNavIcon()}
    </div>
  );
};

export default NavNotExpanded;
