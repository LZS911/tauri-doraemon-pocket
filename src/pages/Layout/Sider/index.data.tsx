import { NavigateFunction } from 'react-router-dom';
import { I18nKey } from '../../../locale';
import { RouterConfigItem } from '../../../router';
import { routerConfig } from '../../../router/router.config';
import { INavIconList } from './index';

export const genNavIconList: (
  navigate: NavigateFunction
) => Array<INavIconList | null> = (navigate) => {
  return routerConfig.map((v) => {
    if (v.menuType === 'Hidden') {
      return null;
    }
    if (!v.children) {
      return {
        name: v.key,
        title: v.label as I18nKey,
        path: v.path as string,
        icon: v.icon,
        onClick: () => {
          navigate(v.path as string);
        },
      };
    }

    return {
      name: v.key,
      path: v.path as string,
      title: v.label as I18nKey,
      icon: v.icon,
      children: v.children.map((c: RouterConfigItem) => {
        return {
          name: c.key,
          path: c.path as string,
          title: c.label as I18nKey,
          onClick: () => {
            navigate(c.path as string);
          },
        };
      }),
    };
  });
};
