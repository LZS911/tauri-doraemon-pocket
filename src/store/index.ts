import { configureStore, Store } from '@reduxjs/toolkit';
import userConfig from './userConfig';
import layout from './layout';
const store: Store = configureStore({
  reducer: {
    userConfig,
    layout,
  },
});
export type IReduxState = ReturnType<typeof store.getState>;
export default store;
