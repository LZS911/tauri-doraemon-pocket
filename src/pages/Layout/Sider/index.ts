import React from 'react';
import { I18nKey } from '../../../locale';
import Sider from './Sider';
export interface INavIconList {
  name: string;
  path: string;
  title: I18nKey;
  icon?: React.ReactNode;
  onClick?: () => void;
  children?: Array<Omit<INavIconList, 'icon' | 'children'>>;
}

export default Sider;
