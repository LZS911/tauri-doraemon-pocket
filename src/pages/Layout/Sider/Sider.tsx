import { MenuFoldOutlined, MenuUnfoldOutlined } from '@ant-design/icons';
import { Drawer } from 'antd';
import classNames from 'classnames';
import { useMemo } from 'react';
import CONSTANT from '../../../common/constant';
import ThemeBase from '../../../components/ThemeBase';
import useLayoutRedux from '../useLayoutRedux';
import NavExpanded from './NavExpanded';
import NavNotExpanded from './NavNotExpanded';

const Sider: React.FC = () => {
  const { isExpandSider, isDrawerSider, setIsExpandSider } = useLayoutRedux();

  const closeDrawerSider = () => {
    setIsExpandSider(false);
  };

  const siderWidth = useMemo(() => {
    if (isDrawerSider) {
      return 0;
    }

    if (isExpandSider) {
      return CONSTANT.SIDER_WIDTH_WHEN_EXPAND;
    }

    return CONSTANT.SIDER_WIDTH;
  }, [isDrawerSider, isExpandSider]);

  const handleClickMenuIcon = (isExpand: boolean) => {
    setIsExpandSider(!isExpand);
  };
  return (
    <nav
      className="sticky top-0 flex min-h-full flex-col items-center justify-center overflow-y-auto pb-3 pl-1 transition-[width]"
      style={{
        width: siderWidth,
      }}
    >
      {isDrawerSider ? (
        <Drawer
          closable={false}
          placement="left"
          visible={isExpandSider}
          onClose={closeDrawerSider}
          bodyStyle={{
            padding: 0,
            width: CONSTANT.SIDER_WIDTH_WHEN_EXPAND,
          }}
          contentWrapperStyle={{
            width: CONSTANT.SIDER_WIDTH_WHEN_EXPAND,
          }}
        >
          <ThemeBase.Paper className="h-full w-full">
            <NavExpanded />
          </ThemeBase.Paper>
        </Drawer>
      ) : (
        <ThemeBase.Paper className="h-full w-full">
          {isExpandSider ? <NavExpanded /> : <NavNotExpanded />}
        </ThemeBase.Paper>
      )}

      <ThemeBase.Icon
        data-testid="menu-icon"
        icon={
          isExpandSider ? (
            <MenuFoldOutlined className="text-[1rem]" />
          ) : (
            <MenuUnfoldOutlined className="text-[1rem]" />
          )
        }
        onClick={() => handleClickMenuIcon(isExpandSider)}
        className={classNames(
          {
            'w-[40%]': !isExpandSider,
            'w-[20%]': isExpandSider,
          },
          'bg-slate-100 dark:bg-black'
        )}
      />
    </nav>
  );
};

export default Sider;
