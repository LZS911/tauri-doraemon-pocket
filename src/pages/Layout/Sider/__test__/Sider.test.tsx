import { waitFor } from '@testing-library/react';
import { useDispatch, useSelector } from 'react-redux';
import { layoutInitialState } from '../../../../store/layout';
import { userConfigInitialState } from '../../../../store/userConfig';
import { renderWithRouter } from '../../../../testUtils/customRender';
import Sider from '../index';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  };
});

describe('test Layout/Sider', () => {
  const useSelectorMock: jest.Mock = useSelector as jest.Mock;
  const useDispatchMock: jest.Mock = useDispatch as jest.Mock;

  beforeEach(() => {
    jest.useFakeTimers();
    useDispatchMock.mockImplementation(() => jest.fn());
    useSelectorMock.mockImplementation((selector) => {
      return selector({
        layout: layoutInitialState,
        userConfig: userConfigInitialState,
      });
    });
  });
  afterEach(() => {
    jest.useRealTimers();
  });

  test('should match snapshot', () => {
    const { container } = renderWithRouter(<Sider />);
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when modified isDrawerSider', async () => {
    useSelectorMock.mockImplementation((selector) => {
      return selector({
        layout: {
          ...layoutInitialState,
          isDrawerSider: true,
        },
        userConfig: userConfigInitialState,
      });
    });

    const { container } = renderWithRouter(<Sider />);
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when modified isExpandSider', async () => {
    useSelectorMock.mockImplementation((selector) => {
      return selector({
        layout: {
          ...layoutInitialState,
          isExpandSider: true,
        },
        userConfig: userConfigInitialState,
      });
    });

    const { container } = renderWithRouter(<Sider />);
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(container).toMatchSnapshot();
  });
});
