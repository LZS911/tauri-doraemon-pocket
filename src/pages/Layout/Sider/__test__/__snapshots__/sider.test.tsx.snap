// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Layout/Sider should match snapshot 1`] = `
<div>
  <nav
    class="sticky top-0 h-full overflow-y-auto transition-[width]"
    style="width: 60px;"
  >
    <div
      class="bg-white dark:bg-darkMode dark:text-white h-full w-full"
    >
      <div
        class="flex flex-col"
      >
        <img
          class="my-2 mx-auto h-10 w-10 cursor-pointer"
          data-testid="logo-img"
          src="/static/images/logo.svg"
          title="logo-img"
        />
        <div
          class="p-3"
        >
          <div
            class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show text-xl dark:bg-darkMode transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
            data-testid="dashboard"
          >
            <span
              aria-label="dashboard"
              class="anticon anticon-dashboard"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="dashboard"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="p-3"
        >
          <div
            class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show text-xl dark:bg-darkMode transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
            data-testid="chat"
          >
            <span
              aria-label="message"
              class="anticon anticon-message"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="message"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M464 512a48 48 0 1096 0 48 48 0 10-96 0zm200 0a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0zm661.2-173.6c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="p-3"
        >
          <div
            class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show text-xl dark:bg-darkMode transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
            data-testid="component"
          >
            <span
              aria-label="code-sandbox"
              class="anticon anticon-code-sandbox"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="code-sandbox"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M709.6 210l.4-.2h.2L512 96 313.9 209.8h-.2l.7.3L151.5 304v416L512 928l360.5-208V304l-162.9-94zM482.7 843.6L339.6 761V621.4L210 547.8V372.9l272.7 157.3v313.4zM238.2 321.5l134.7-77.8 138.9 79.7 139.1-79.9 135.2 78-273.9 158-274-158zM814 548.3l-128.8 73.1v139.1l-143.9 83V530.4L814 373.1v175.2z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`test Layout/Sider should match snapshot when modified isDrawerSider 1`] = `
<div>
  <nav
    class="sticky top-0 h-full overflow-y-auto transition-[width]"
    style="width: 0px;"
  />
</div>
`;

exports[`test Layout/Sider should match snapshot when modified isExpandSider 1`] = `
<div>
  <nav
    class="sticky top-0 h-full overflow-y-auto transition-[width]"
    style="width: 260px;"
  >
    <div
      class="bg-white dark:bg-darkMode dark:text-white h-full w-full"
    >
      <div
        class="flex flex-col"
      >
        <div
          class="flex h-14 items-center pl-4"
        >
          <img
            class="mr-3"
            src="/static/images/logo.svg"
            title="logo"
          />
          <span
            class="text-2xl text-black dark:text-white"
          >
            login.loginHeader.title
          </span>
        </div>
        <ul
          class="ant-menu ant-menu-root ant-menu-inline ant-menu-light theme-menu text-base"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        >
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/dashboard"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="dashboard"
              class="anticon anticon-dashboard ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="dashboard"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              router.title.dashboard
            </span>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/chat"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="message"
              class="anticon anticon-message ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="message"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M464 512a48 48 0 1096 0 48 48 0 10-96 0zm200 0a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0zm661.2-173.6c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              router.title.chat
            </span>
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="none"
          >
            <div
              aria-controls="rc-menu-uuid-test-/component-popup"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              data-menu-id="rc-menu-uuid-test-/component"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                aria-label="code-sandbox"
                class="anticon anticon-code-sandbox ant-menu-item-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="code-sandbox"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M709.6 210l.4-.2h.2L512 96 313.9 209.8h-.2l.7.3L151.5 304v416L512 928l360.5-208V304l-162.9-94zM482.7 843.6L339.6 761V621.4L210 547.8V372.9l272.7 157.3v313.4zM238.2 321.5l134.7-77.8 138.9 79.7 139.1-79.9 135.2 78-273.9 158-274-158zM814 548.3l-128.8 73.1v139.1l-143.9 83V530.4L814 373.1v175.2z"
                  />
                </svg>
              </span>
              <span
                class="ant-menu-title-content"
              >
                router.title.component
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
          </li>
        </ul>
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </div>
    </div>
  </nav>
</div>
`;
